### PR TITLE
fix(ui): fix clipboardData requires non-nullable string as text.

### LINKED ISSUE #1529: Flutter 3.10 Build error
### Which packages are you using?

stream_chat_flutter

### On what platforms did you experience the issue?

iOS, Android

### What version are you using?

stream_chat_flutter: ^6.0.0

### What happened?

Could not build the precompiled application for the device.
Error (Xcode): ../../.pub-cache/hosted/pub.dev/stream_chat_flutter-6.0.0/lib/src/message_widget/message_widget.dart:1111:81: Error: The argument type 'String?' can't be assigned to the parameter type 'String' because 'String?' is nullable and 'String' isn't.

### Steps to reproduce

```bash
1. flutter run
```

### Supporting info to reproduce

_No response_

### Relevant log output

```shell
Could not build the precompiled application for the device.
Error (Xcode): ../../.pub-cache/hosted/pub.dev/stream_chat_flutter-6.0.0/lib/src/message_widget/message_widget.dart:1111:81: Error: The argument type 'String?' can't be assigned to the parameter type 'String' because 'String?' is nullable and 'String' isn't.
```

### Flutter analyze output

_No response_

### Flutter doctor output

```shell
[✓] Flutter (Channel stable, 3.10.0, on macOS 13.3.1 22E772610a darwin-arm64, locale en-GB)
    • Flutter version 3.10.0 on channel stable at /Users/christopherelliot/Develop/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 84a1e904f4 (2 days ago), 2023-05-09 07:41:44 -0700
    • Engine revision d44b5a94c9
    • Dart version 3.0.0
    • DevTools version 2.23.1

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.2)
    • Android SDK at /Users/christopherelliot/Library/Android/sdk
    • Platform android-33, build-tools 33.0.2
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14E222b
    • CocoaPods version 1.12.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2022.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)

[✓] VS Code (version 1.78.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.64.0

[✓] Connected device (3 available)
    • iPhone (91) (mobile) • 00008110-000918AE14D0401E • ios            • iOS 16.4.1 20E252
    • macOS (desktop)      • macos                     • darwin-arm64   • macOS 13.3.1 22E772610a darwin-arm64
    • Chrome (web)         • chrome                    • web-javascript • Google Chrome 113.0.5672.92
    ! Error: Christopher’s Apple Watch has not finished loading development services. Please select a different device, or wait for the device to load development services and try again. (code 8)

[✓] Network resources
    • All expected network resources are available.

• No issues found!
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by ChrisElliotUK**
# Submit a pull request

## CLA

- [X] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required).
- [X] The code changes follow best practices
- [X] Code changes are tested (add some information if not applicable)

## Description of the pull request
Fixes #1529: ClipboardDate text param requires non nullable string.

### CODE REVIEW

File: packages/stream_chat_flutter/lib/src/message_widget/message_widget.dart [hunk 1]
```diff
@@ -1009,7 +1009,7 @@ class _StreamMessageWidgetState extends State<StreamMessageWidget>
           title: Text(context.translations.copyMessageLabel),
           onClick: () {
             Navigator.of(context, rootNavigator: true).pop();
-            Clipboard.setData(ClipboardData(text: widget.message.text));
+            Clipboard.setData(ClipboardData(text: widget.message.text ?? ''));
```
Comment (xsahil03x) [inline packages/stream_chat_flutter/lib/src/message_widget/message_widget.dart:1012]: Can you instead check for the text to be non-null?

```suggestion
            final text = message.text;
            if (text != null) Clipboard.setData(ClipboardData(text: text));
```

File: packages/stream_chat_flutter/lib/src/message_widget/message_widget.dart [hunk 2]
```diff
@@ -1169,7 +1169,7 @@ class _StreamMessageWidgetState extends State<StreamMessageWidget>
                     : DisplayWidget.show,
           ),
           onCopyTap: (message) =>
-              Clipboard.setData(ClipboardData(text: message.text)),
+              Clipboard.setData(ClipboardData(text: message.text ?? '')),
```
Comment (xsahil03x) [inline packages/stream_chat_flutter/lib/src/message_widget/message_widget.dart:1172]: Same thing here.

```dart
final text = message.text;
            if (text != null) Clipboard.setData(ClipboardData(text: text));
```

Comment (ChrisElliotUK) [reply packages/stream_chat_flutter/lib/src/message_widget/message_widget.dart]: Done

File: packages/stream_chat_flutter/CHANGELOG.md [hunk 3]
```diff
@@ -15,6 +15,7 @@
   applied correctly.
 - [[#1525]](https://github.com/GetStream/stream-chat-flutter/issues/1525) Fixed `StreamQuotedMessageWidget` message for
   deleted messages not being shown correctly.
+- [[#1529]](https://github.com/GetStream/stream-chat-flutter/issues/1529) Fixed fix `ClipboardData` requires non-nullable string as text on Flutter 2.10.
```
Comment (xsahil03x) [inline packages/stream_chat_flutter/CHANGELOG.md:18]:
```suggestion
- [[#1529]](https://github.com/GetStream/stream-chat-flutter/issues/1529) Fixed fix `ClipboardData` requires non-nullable string as text on Flutter 3.10. 
```

File: packages/stream_chat_flutter/lib/src/message_widget/message_widget.dart [hunk 4]
```diff
@@ -1168,8 +1169,10 @@ class _StreamMessageWidgetState extends State<StreamMessageWidget>
                     ? DisplayWidget.gone
                     : DisplayWidget.show,
           ),
-          onCopyTap: (message) =>
-              Clipboard.setData(ClipboardData(text: message.text)),
+          onCopyTap: (message) {
+             final text = message.text;
```
Comment (xsahil03x) [inline packages/stream_chat_flutter/lib/src/message_widget/message_widget.dart:1173]:
```suggestion
            final text = message.text;
```

File: packages/stream_chat_flutter/lib/src/message_widget/message_widget.dart [hunk 5]
```diff
@@ -1009,7 +1009,8 @@ class _StreamMessageWidgetState extends State<StreamMessageWidget>
           title: Text(context.translations.copyMessageLabel),
           onClick: () {
             Navigator.of(context, rootNavigator: true).pop();
-            Clipboard.setData(ClipboardData(text: widget.message.text));
+            final text = message.text;
```
Comment (xsahil03x) [inline packages/stream_chat_flutter/lib/src/message_widget/message_widget.dart:1012]:
```suggestion
            final text = widget.message.text;
```

**Review by xsahil03x — APPROVED** [review]

**Pull request merged**